include/openthread/ip6.h

```diff
@@ -114,6 +114,15 @@ OTAPI otError OTCALL otIp6RemoveUnicastAddress(otInstance *aInstance, const otIp
 OTAPI const otNetifAddress *OTCALL otIp6GetUnicastAddresses(otInstance *aInstance);
 
 /**
+ * The function returns the link-local IPv6 address assigned to the Thread interface.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ *
+ * @returns A pointer to link-local Network Interface Address, or NULL if there is no link-local address.
+ */
+const otNetifAddress *otIp6GetLinkLocalAddress(otInstance *aInstance);
```

Comment:
This assumes that a Thread interface will have exactly one link-local address.  While this may be true in most cases, I'm not sure we can assume this in the general case.